apps/meteor/client/views/hooks/useMembersList.ts

```diff
@@ -11,12 +14,108 @@ type MembersListOptions = {
 
 const endpointsByRoomType = {
 	d: '/v1/im.members',
-	p: '/v1/groups.members',
-	c: '/v1/channels.members',
+	p: '/v1/rooms.membersOrderedByRole',
+	c: '/v1/rooms.membersOrderedByRole',
 } as const;
 
+type RoomMember = Pick<IUser, 'username' | '_id' | 'name' | 'status' | 'freeSwitchExtension'> & { roles?: IRole['_id'][] };
+
+const getSortedMembers = (members: RoomMember[], useRealName = false) => {
+	return members.sort((a, b) => {
+		const aRoles = a.roles ?? [];
+		const bRoles = b.roles ?? [];
+		const isOwnerA = aRoles.includes('owner');
+		const isOwnerB = bRoles.includes('owner');
+		const isModeratorA = aRoles.includes('moderator');
+		const isModeratorB = bRoles.includes('moderator');
+
+		if (isOwnerA !== isOwnerB) {
+			return isOwnerA ? -1 : 1;
+		}
+
+		if (isModeratorA !== isModeratorB) {
+			return isModeratorA ? -1 : 1;
+		}
+
+		if (a.status === 'online' && b.status !== 'online') {
+			return -1;
+		}
+
+		if (a.status !== 'online' && b.status === 'online') {
+			return 1;
+		}
```

Comment:
```suggestion
		if ((a.status === 'online' || b.status === 'online') && a.status !== b.status) {
			return a.status === 'online' ? -1 : 1;
		}
```
wdyt? maybe readability is better the way you wrote it, so feel free to resolve this if you prefer as it is